Builtin support for SQLite schema table

trueqbit opened the pull request:
Ability to add the table "sqlite_master" to a storage and select from it or its alias "sqlite_schema".

Additionally:
* Ability to "get all" from an aliased table

Still open for discussion:
1. Despite the fact that the schema table was renamed from "sqlite_master" to "sqlite_schema" in SQLite 3.33.0, the renaming process was more like keeping the previous name "sqlite_master" and appending an internal alias "sqlite_schema".
  This fact can be deduced from the following:
  If the set of columns is qualified, the following select statement will fail with the error "no such table: sqlite_schema": `SELECT sqlite_schema.* from sqlite_schema`.
  So I suggest keeping the previous table name "sqlite_master" and providing "sqlite_schema" as a table alias in sqlite_orm.
2. Was `get_all<Object>()` intended to be shorthand for `select(object<Object>())`? If so, it makes sense to provide overloads for selecting from an alias table. Otherwise I don't want to fall into over-engineering.

fnc12 commented on tests/tests3.cpp:
```diff
@@ -121,6 +121,19 @@ TEST_CASE("Wide string") {
 }
 #endif  //  SQLITE_ORM_OMITS_CODECVT
+
+TEST_CASE("sqlite_schema") {
```
idea for reducing amount of tests inside `tests*.cpp` files: we can move `TEST_CASE("sqlite_schema")` and `TEST_CASE("dbstat")` into a dedicated file called how you call such kind of tables

trueqbit commented:
I packed a bunch of other code quality improvements into this PR because they popped up while I was working on it. Most notable:
* Corrected EOL sequence in header files to LF
* unit tests: Corrected decomposition for expressions required to be false
* appveyor: Enabled unit testing SQLite json1 functionality

trueqbit commented on dev/statement_serializer.h:
```diff
@@ -1350,13 +1350,20 @@ namespace sqlite_orm {
-               << streaming_identifier(table.name) << streaming_conditions_tuple(getAll.conditions, context);
+            ss << "SELECT "
+               << streaming_table_column_names(table,
+                                               is_table_alias_v<table_type> ? alias_extractor<table_type>::as_alias()
```
Moved into `alias_extractor<>::as_qualifier()`
